packages/liveblocks-core/src/http-client.ts

```diff
@@ -7,3 +22,117 @@ export function getBearerTokenFromAuthValue(authValue: AuthValue): string {
     return authValue.token.raw;
   }
 }
+
+/**
+ * @internal
+ *
+ * Small HTTP client for client-only REST API requests (e.g. /v2/c/* URLs).
+ * These URLs all use public key, ID token, or access token authorization. This
+ * HTTP client can be shared and used by both the Liveblocks Client and
+ * Liveblocks Room instances internally to talk to our client-only REST API
+ * backend.
+ */
+export class HttpClient {
+  private _baseUrl: string;
+  private _authCallback: () => Promise<AuthValue>;
+  private _fetchPolyfill: typeof fetch;
+
+  constructor(
+    baseUrl: string,
+    fetchPolyfill: typeof fetch,
+    authCallback: () => Promise<AuthValue>
+  ) {
+    this._baseUrl = baseUrl;
+    this._fetchPolyfill = fetchPolyfill;
+    this._authCallback = authCallback;
+  }
+
+  // ------------------------------------------------------------------
+  // Public methods
+  // ------------------------------------------------------------------
+
+  /**
+   * Constructs and makes the HTTP request, but does not handle the response.
+   *
+   * This is what .fetch() does:    👈 This method!
+   *   1. Set Content-Type header
+   *   2. Set Authorization header
+   *   3. Call the callback to obtain the `authValue` to use in the Authorization header
+   *
+   * This is what .fetchJson() does ON TOP of that:
+   *   4. Parse response body as Json
+   *   5. ...but silently return `{}` if that parsing fails
+   *   6. Throw HttpError if response is an error
+   */
+  // XXX Ultimately, might be nice to make this a private method? It might be much nicer if this method would _always_ deal with the JSON parsing.
+  public async fetch(
+    endpoint: URLSafeString,
+    options?: RequestInit,
+    params?: QueryParams
+  ): Promise<Response> {
```

Comment:
Like you mentioned, it'd be ideal for all methods to be using the `fetchJson()` method and `fetch` to be an implementation detail, but there can be some breaking changes if we do that change in a rush.